docs/frontend-system/building-plugins/testing.md

```diff
@@ -0,0 +1,237 @@
+---
+id: testing
+title: Frontend System Testing Plugins
+sidebar_label: Testing
+# prettier-ignore
+description: Testing plugins in the frontend system
+---
+
+> **NOTE: The new frontend system is in a highly experimental phase**
+
+# Testing Frontend Plugins
+
+> NOTE: The new frontend system is in alpha, and some plugins do not yet fully implement it.
+
+Utilities for testing frontend features and components are available in `@backstage/frontend-test-utils`.
+
+## Testing components
+
+A component can be used for more than one extension, and it should be tested independently of an extension environment.
+
+Use the `renderInTestApp` helper to render a given component inside a Backstage test app:
+
+```tsx
+import React from 'react';
+import { screen } from '@testing-library/react';
+import { renderInTestApp } from '@backstage/frontend-test-utils';
+import { EntityDetails } from './plugin';
+
+describe('Entity details component', () => {
+  it('should render the entity name and owner', async () => {
+    renderInTestApp(<EntityDetails owner="tools" name="test" />);
+
+    await expect(
+      screen.getByText('The entity "test" is owned by "tools"'),
+    ).resolves.toBeInTheDocument();
+  });
+});
+```
+
+It's important to highlight that mocking APIs for components is different from mocking them for extensions. In the snippet below, we wrapped the component within a `TestApiProvider` for mocking the Catalog API:
```

Comment:
```suggestion
To mock [Utility APIs](../utility-apis/01-index.md) that are used by your component you can use the `TestApiProvider` to override individual API implementations. In the snippet below, we wrap the component within a `TestApiProvider` in order to mock the catalog client API:
```

Thinking we can keep this straight to the point of how to mock APIs for components and worry about extensions later.